TsRandomizer/LevelObjects/ItemManipulators/JournalMemoryEvent+JournalLetterEvent.cs

```diff
@@ -0,0 +1,30 @@
+﻿using Timespinner.GameObjects.BaseClasses;
+using TsRandomizer.IntermediateObjects;
+using TsRandomizer.Randomisation;
+using TsRandomizer.Screens;
+
+namespace TsRandomizer.LevelObjects.ItemManipulators
+{
+	[TimeSpinnerType("Timespinner.GameObjects.Events.Treasure.JournalLetterEvent")]
+	[TimeSpinnerType("Timespinner.GameObjects.Events.Treasure.JournalMemoryEvent")]
+	// ReSharper disable once UnusedMember.Global
+	class MemoryEvent : ItemManipulator
+	{
+		bool hasAwardedItem;
+
+		public MemoryEvent(Mobile typedObject, ItemLocation itemLocation) : base(typedObject, itemLocation)
+		{
+		}
+
+		protected override void OnUpdate(GameplayScreen gameplayScreen)
+		{
+			Dynamic._oscillationDelta = 0; // Fix the item's position so the replacement doesn't miss
```

Comment:
i dont think thr oscilation should make replacement fail as we replace it always on the same frame,

Reply:
I was consistently seeing it fail without this. Not sure quite _why_ since as you said it should just replace on screen change, and I didn't change any of the x/y positions before this fix made all of them consistently work.

I do want to look into this some more for a future pass, but for this first PR I'd like to leave it in, because it would take me a lot of attempts to be 100% sure the bug was fixed, and this still spins/animates/plays sound, it just doesn't bob up and down.

Reply:
(It's a quirk similar to the timespinner piece offset, but unlike those which are replaced by a dropped-on-the-ground item, I'd like to leave these as floating mystery checks, like how terminals are surprise items)